chapter_1/exercise_1_22/fold_line.c

```diff
@@ -43,30 +43,44 @@ int get_line(char line[], int max_line_len)
 
 void fold_line(char line[], char fold_str[], int n_break)
 {
-  int i, j;
+  int i, j, m;
+  int column = 0;
   int split = FALSE;
   int offset = 10;
 
-  for (i = 0, j = 0; line[i] != '\0'; ++i, ++j)
+  for (i = 0, j = 0; line[i] != '\0'; i++, j++)
   {
-    if (i && i % n_break == 0)
-    {
-      split = TRUE;
-    }
+    fold_str[j] = line[i];
 
-    if (split && (line[i] == ' ' || i > n_break + offset))
+    if (fold_str[j] == '\n')
     {
-      fold_str[j++] = '\n';
-      split = FALSE;
-      ++i;
+      column = 0;
     }
 
-    fold_str[j] = line[i];
-  }
+    column++;
 
-  if (line[i] == '\n')
-  {
-    fold_str[j++] = line[i];
+    if (column == n_break)
+    {
+      split = TRUE;
+      for (m = j; m >= j - offset && split; m--)

```

Comment:
@illustrofia The same think applies here with the prefixed incrementation of the m variable.